### PR TITLE
Added current git commit to status page. Refactorings. Decided to not add Persons.birth_date.

### PR DESCRIPTION
**Pull request by larspetrus**
I intended to add a `date` column to `Persons`, like we did for `Competitions`. But after looking at the usage I don't think it's needed. The `Persons.year/month/day` columns are just data to display and is never used in queries, so what's there works fine. There are 3 other tables with year/month/day columns, but they seem to be all PHP.

What I did do:
- Fixed all remaining usages of `Companies.year/month/day`.
- Added link to the git commit actually running to `/server-status` page. Hope it works in our setup :)
- Did the most obvious and safe refactorings in the files I touched.

### CODE REVIEW

File: WcaOnRails/app/views/server_status/index.html.erb [hunk 1]
```diff
@@ -62,6 +62,8 @@
     <span>PID:</span> <%= Process.pid %>
     <%= icon "circle" %>
     <span>Uptime:</span> <%= duration_to_s(Time.now - WcaOnRails::BOOTED_AT) %>
+    <%= icon "circle" %>
+    <a href="https://github.com/thewca/worldcubeassociation.org/commit/#{`git rev-parse HEAD`}">Git commit</a>
   <% end %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/server_status/index.html.erb]: See [a part of this Slack conversation](https://worldcubeassociation.slack.com/archives/i18n/p1487366833925354). I'd suggest something like this
```erb
  <%= alert :info do
    sha1 = `git rev-parse HEAD`
    {
      "PID": Process.pid,
      "Uptime": duration_to_s(Time.now - WcaOnRails::BOOTED_AT),
      "Git commit": link_to(sha1.first(7), "https://github.com/thewca/worldcubeassociation.org/commit/#{sha1}")
    }.map { |key, value| "#{key}: #{value}" }.join(" #{icon "circle"} ").html_safe
  end %>
```
If it's still an overkill in your opinion, I am fine with this code =)

Comment (larspetrus) [inline WcaOnRails/app/views/server_status/index.html.erb]: That is a clear improvement! Especially that you can see the short sha on the page. The lack of ERB syntax debris is also real nice.

**Comment by larspetrus** [comment]
Yeah, I don't really know what I'm doing with all the merge and push options.

I get things to work, but it's probably not always best practice.

Any tips?

**Comment by jonatanklosko** [comment]
@larspetrus I always pull the to the local *master* branch and then do `rebase master` from the branch I work on, resolving any conflicts. This assumes that you work on a separate branch of course. Note that then you have to force push the changes (which is actually fine as you're the only one working on the branch) and both changes and history stays clean.

I'm not certain how to clean the history of this PR, so maybe it would be ok to let it remain as it is.

LGTM

**Comment by larspetrus** [comment]
I used the **Squash and merge** button on Github, and it looks like it did something reasonable.

**Comment by jonatanklosko** [comment]
Oh, Github squashed everything into one commit on its own, that's neat!

**Pull request merged**